services/ymax-planner/src/sequence-manager.ts

```diff
@@ -0,0 +1,108 @@
+import { Fail } from '@endo/errors';
+import type { CosmosRestClient } from './cosmos-rest-client.ts';
+
+type AccountInfo = {
+  '@type': string;
+  address: string;
+  pub_key?: {
+    '@type': string;
+    key: string;
+  };
+  account_number: string;
+  sequence: string;
+};
+
+type AccountResponse = {
+  account: AccountInfo;
+};
+
+type SequenceManagerConfig = {
+  address: string;
+  chainKey: string;
+};
+
+type SequenceManagerPowers = {
+  cosmosRest: CosmosRestClient;
+  log?: (...args: unknown[]) => void;
+};
+
+export class SequenceManager {
```

Comment:
I don't have much experience with JS classes. I defer to @gibson042

Reply:
FYI `private` is smoke from TS that doesn't actually create private fields. Why not use actual JS private fields, aka `#foo` ?

Using a class is "ok" if we maintain proper encapsulation.

Reply:
> Using a class is "ok" if we maintain proper encapsulation.

Note that this means using actual JS `#foo` private fields and not TS `private foo`.